unittests in test_fpp.py

adenjonah opened the pull request:
I created tests for all non-graphical functions in fpp and verified intended output.

I used assertAlmostEqual due to Python decimal arithmetic.

I also added an if name == main at the end of this test file to allow for easier running of the tests from a terminal.

Lastly, I had to create a virtual environment to get statwrap working locally on my machine, so I added all newly created venv and cache files to the gitignore.

alexanderthclark commented on tests/test_fpp.py:
```diff
@@ -1,20 +1,242 @@
+        expected = 0.2
+        self.assertAlmostEqual(result, expected)
+
+    def test_average_multiple_args(self):
```
can you include a comment to make the difference between this and the test on 43 more apparent, where this one unpacks a list and the other doesn't?

alexanderthclark commented on tests/test_fpp.py:
```diff
@@ -1,20 +1,242 @@
+        self.numbers_multiple = [1, 2, 3, 4, 5]
+        self.single_value = [42]
+        self.mixed_types = [1, 2.5, 3, 4.5]
+        self.zeroes = [0, 0, 0, 0]
```
let's add a test for `self.singleton = [1]` where that should give a zerodivision or another error.

alexanderthclark reviewed (changes requested):
Looks good! Just a couple comments. 

edit: Can you also just confirm in the PR that you were able to successfully install the package locally?

adenjonah commented:
<img width="685" alt="image" src="https://github.com/alexanderthclark/statwrap/assets/151240959/f7c95f0b-aefb-44ad-a337-3513f1c66f8c">
<img width="687" alt="image" src="https://github.com/alexanderthclark/statwrap/assets/151240959/d25515ea-d438-4789-969f-f680264219b3">

- Added comment
- Made vars_plus and sd_plus raise an error when n <= 1
- Rewrote documentation to reflect that change (pictured above)
- Added tests for testing the error functionality
- These changes are in the following commit

adenjonah commented:
Lastly, to confirm I was able to install the package locally

alexanderthclark reviewed (approved):
lgtm
